pkg/cmd/pipeline/describe.go

```diff
@@ -66,70 +90,38 @@ func printPipelineDescription(out io.Writer, p cli.Params, pname string) error {
 		return err
 	}
 
-	w := tabwriter.NewWriter(out, 0, 5, 3, ' ', tabwriter.TabIndent)
-	fmt.Fprintf(out, "Name: %s\n", pname)
-	printResources(w, pipeline)
-	printTasks(w, pipeline)
-	printRuns(w, cs, p, pname)
-
-	return w.Flush()
-}
-
-func printResources(w *tabwriter.Writer, pipeline *v1alpha1.Pipeline) {
-	fmt.Fprintln(w, "\nResources")
-	if len(pipeline.Spec.Resources) == 0 {
-		fmt.Fprintln(w, "No resources found")
-		return
-	}
-	fmt.Fprintf(w, resHeader)
-	for _, val := range pipeline.Spec.Resources {
-		fmt.Fprintf(w, resBody,
-			val.Name,
-			val.Type,
-		)
-
+	opts := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("tekton.dev/pipeline=%s", pname),
 	}
-}
-
-func printTasks(w *tabwriter.Writer, pipeline *v1alpha1.Pipeline) {
-	fmt.Fprintln(w, "\nTasks")
-	if len(pipeline.Spec.Tasks) == 0 {
-		fmt.Fprintln(w, "No tasks found")
-		return
+	pipelineRuns, err := cs.Tekton.TektonV1alpha1().PipelineRuns(p.Namespace()).List(opts)
+	if err != nil {
+		return err
 	}
-	fmt.Fprintf(w, taskHeader)
-	for _, val := range pipeline.Spec.Tasks {
-		fmt.Fprintf(w, taskBody,
-			val.Name,
-			val.TaskRef.Name,
-			val.RunAfter,
-		)
 
+	var data = struct {
+		Pipeline     *v1alpha1.Pipeline
+		PipelineRuns *v1alpha1.PipelineRunList
+		PipelineName string
+		Params       cli.Params
+	}{
+		Pipeline:     pipeline,
+		PipelineRuns: pipelineRuns,
+		PipelineName: pname,
+		Params:       p,
 	}
-}
 
-func printRuns(w *tabwriter.Writer, cs *cli.Clients, p cli.Params, pname string) {
-	fmt.Fprintln(w, "\nRuns")
-	opts := metav1.ListOptions{
-		LabelSelector: fmt.Sprintf("tekton.dev/pipeline=%s", pname),
+	funcMap := template.FuncMap{
+		"formatAge":       formatted.Age,
+		"formatDuration":  formatted.Duration,
+		"formatCondition": formatted.Condition,
 	}
 
-	runs, err := cs.Tekton.TektonV1alpha1().PipelineRuns(p.Namespace()).List(opts)
+	w := tabwriter.NewWriter(out, 0, 5, 3, ' ', tabwriter.TabIndent)
+	t := template.Must(template.New("Describe Pipeline").Funcs(funcMap).Parse(templ))
+	err = t.Execute(w, data)
 	if err != nil {
-		fmt.Fprintln(os.Stderr, err)
-	}
-	if len(runs.Items) == 0 {
-		fmt.Fprintln(w, "No runs found")
-		return
+		panic(err)
```

Comment:
Any reason for panic instead of return an error? :thinking:

Reply:
There are 2 places where it panics 
- `template.Must` 
- within `template.Execute` see: errRecover 

But errRecover does return/set the err in some cases, where we could just exit by writing `StdErr` (use cli.Stream)

Reply:
How about https://github.com/tektoncd/cli/pull/132

Reply:
Thanks for the review! I fixed to simply return an error instead of panicing!

Reply:
Thanks!